src/jvm.rs

```diff
@@ -125,6 +133,100 @@ fn get_or_default_init_jvm() -> crate::GlobalResult<JvmPtr> {
     }
 }
 
+/// Invoked as the body from a JNI native function when it is called by the JVM.
+/// Initializes the environment and invokes `op`. Converts the result into a java
+/// object and returns it. Caller should then return this to the JVM.
+///
+/// # Safety condition
+///
+/// Must be invoked as the entire body of a JNI native function, with
+/// `env` being the `EnvPtr` argument provided.
+pub unsafe fn native_function_returning_object<J, R>(
+    env: EnvPtr<'_>,
+    op: impl FnOnce() -> R,
+) -> jni_sys::jobject
+where
+    J: Upcast<crate::java::lang::Object> + Upcast<J>,
+    R: ToJavaImpl<J> + std::fmt::Debug,
```

Comment:
Do you still need the `Debug`?
```suggestion
    R: ToJavaImpl<J>,
```

Reply:
Oh, I do not, good catch.